csa.rs

```diff
@@ -25,24 +25,22 @@ impl Connection {
 }
 
 fn csa_main_loop(timetable: &[Connection], arrival_station: usize, earliest_arrival: &mut [u32], in_connection: &mut [usize]) {
-    let mut earliest = std::u32::MAX;
-
-    for (i, connection) in timetable.iter().enumerate() {
+    timetable.iter().enumerate().fold(std::u32::MAX, |earliest, (i, connection)| {
```

Comment:
Using fold in this case is particularly unnatural. The goal of fold is to pass a value throw a stream using a function, not removing a side effect. If you don't use the result of fold, please don't use it.